src/nvim/eval.c

```diff
@@ -8717,7 +8717,6 @@ static void f_environ(typval_T *argvars, typval_T *rettv, FunPtr fptr)
   tv_dict_alloc_ret(rettv);
 
   for (int i = 0; ; i++) {
-    // TODO(justinmk): use os_copyfullenv from #7202 ?
     char *envname = os_getenvname_at_index((size_t)i);
```

Comment:
@jamessan would it not be better to use `os_copyfullenv` here  and then convert to `rettv->vval.v_dict`? 

Time-cost of calling `os_getenvname_at_index` in a loop is O(n^2).

Reply:
Right, I had thought about that and don't remember why I didn't.